### DISCUSSION TITLE
Need help with bypassing L1D cache for storing Translation data

### DISCUSSION THREAD

**Discussion by ShreyanshNeekhre** [ChampSim/ChampSim#385] [General]
I wanted to bypass L1D cache to direct the page table entries read request and MSHR fill request to directly go to L2C cache from Page Structure Cache. And also wanted to avoid storing Page table entries in the L1D cache.
Please help.

**Comment by ngober** [comment]
It sounds like you want PTW requests to be issued to the L2C. Have you tried configuring with the following config file?

    {
        "ooo_cpu": [{
            "L2C": { "name": "my_l2c" },
            "PTW": { "lower_level": "my_l2c" }
        }]
    }

**Reply by ShreyanshNeekhre** [reply]
no, i didn't find this file, Where can i find this file in Champsim

**Reply by ngober** [reply]
You should read the documentation on the configuration step in ChampSim: https://champsim.github.io/ChampSim/master/Creating-a-configuration-file.html

**Comment by ShreyanshNeekhre** [comment]
I am not able to find this file in ChmapSim.

**Comment by ShreyanshNeekhre** [comment]
I have tried changing the champsim_config.json file but  it is not changing
the simulation results.
I am attaching the screenshot of changes that i have done giving the same
results for both , before and after adding the lines
It should make TRANSLATION ACCESS of L1D CACHE to zero.

**Reply by ngober** [reply]
I do see a change when running this locally. I ran 50M instructions on `server_011`, from the set of IPC1 traces.

When no configuration is specified, the simulation yields 152806 translation accesses to the L1D.

When the above configuration is specified, the simulation yields 152775 translation accesses to the L2C and zero translation accesses to the L1D. This appears to verify that the configuration above redirects translation requests from the L1 to the L2.

**Comment by ShreyanshNeekhre** [comment]
?

**Comment by ShreyanshNeekhre** [comment]
hey thanks for the help, the issue is solved now.
I need one more help,
I am trying to bypass Page table entries from all the caches and trying to
fetch them from the DRAM, by simply changing the lower level of PWC to DRAM
in the config.json file, but the simulations are showing some deadlock
conditions that the IFETCH buffer is throwing , because of one condition
failing i.e
if (!std::empty(IFETCH_BUFFER) && (IFETCH_BUFFER.front().event_cycle +
DEADLOCK_CYCLE) <= current_cycle)
throw champsim::deadlock{cpu};
Any ideas how to resolve this?
